compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TaintAnalyzer.java

```diff
@@ -2445,6 +2473,28 @@ private void analyzeInvocation(BLangInvocation invocationExpr) {
         getCurrentAnalysisState().taintedStatus = returnTaintedStatus;
     }
 
+    private boolean restoreTableIfIgnored(BInvokableSymbol invokableSymbol, Map<Integer, TaintRecord> origTaintTable,
+     TaintedStatus argumentTaintReturnValue) {
+        if (argumentTaintReturnValue != TaintedStatus.IGNORED) {
+            return false;
+        }
+        invokableSymbol.taintTable = origTaintTable; // restore taint table.
+        return true;
+    }
+
+    private TaintedStatus updateAndGetTaintedStatus(BLangInvocation invocationExpr, TaintedStatus returnTaintedStatus,
```

Comment:
Do we need this param `returnTaintedStatus`

Reply:
If `argumentTaintReturnValue == TaintedStatus.TAINTED` evaluates to false, we seem to be using this original `returnTaintedStatus`, so I guess we still need it?

Reply:
ack